Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -372,27 +374,72 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::index_of)
     return index.has_value() ? Value(*index) : Value(-1);
 }
 
-// 22.1.3.24 String.prototype.toLocaleLowerCase ( [ reserved1 [ , reserved2 ] ] ), https://tc39.es/ecma262/#sec-string.prototype.tolocalelowercase
-// NOTE: This is the minimum toLocaleLowerCase implementation for engines without ECMA-402.
+static Optional<String> resolve_best_locale(GlobalObject& global_object, Value locales)
+{
+    // For details on these steps, see https://tc39.es/ecma402/#sup-string.prototype.tolocalelowercase
+    auto& vm = global_object.vm();
+
+    // 3. Let requestedLocales be ? CanonicalizeLocaleList(locales).
+    auto requested_locales = Intl::canonicalize_locale_list(global_object, locales);
+    if (vm.exception())
+        return {};
+
+    Optional<Unicode::LocaleID> requested_locale;
+
+    // 4. If requestedLocales is not an empty List, then
+    if (!requested_locales.is_empty()) {
+        // a. Let requestedLocale be requestedLocales[0].
+        requested_locale = Unicode::parse_unicode_locale_id(requested_locales[0]);
+    }
+    // 5. Else,
+    else {
+        // a. Let requestedLocale be DefaultLocale().
+        requested_locale = Unicode::parse_unicode_locale_id(Unicode::default_locale());
+    }
+    VERIFY(requested_locale.has_value());
+
+    // 6. Let noExtensionsLocale be the String value that is requestedLocale with any Unicode locale extension sequences (6.2.1) removed.
+    requested_locale->remove_extension_type<Unicode::LocaleExtension>();
+    auto no_extensions_locale = JS::Intl::canonicalize_unicode_locale_id(*requested_locale);
```

Comment:
not sure about this... it's essentially adding a third variant of the same step, using a mix of the two existing ones `:thonk:`

https://github.com/SerenityOS/serenity/blob/6ffc8f389e76d8358a8829948dd20f2a5603c8bc/Userland/Libraries/LibJS/Runtime/Intl/LocaleConstructor.cpp#L229-L231

https://github.com/SerenityOS/serenity/blob/6ffc8f389e76d8358a8829948dd20f2a5603c8bc/Userland/Libraries/LibJS/Runtime/Intl/AbstractOperations.cpp#L280-L281

https://github.com/SerenityOS/serenity/blob/6ffc8f389e76d8358a8829948dd20f2a5603c8bc/Userland/Libraries/LibJS/Runtime/Intl/AbstractOperations.cpp#L417-L419

Reply:
Ah, yeah, I should use `to_string` here since it's already canonical.

This one here should also be updated:
https://github.com/SerenityOS/serenity/blob/6ffc8f389e76d8358a8829948dd20f2a5603c8bc/Userland/Libraries/LibJS/Runtime/Intl/AbstractOperations.cpp#L280-L281
It's removing all extensions rather than just Unicode locale extensions (I think I misunderstood the difference at first), and can also use `to_string`